src/lib/converter/comments/textParser.ts

```diff
@@ -311,6 +311,11 @@ function checkTagLink(data: TextParserData): RelativeLink | undefined {
         data.pos += 3;
         return checkAttribute(data, "href");
     }
+
+    if (token.text.startsWith("<source ", pos)) {
+        data.pos += 8;
+        return checkAttribute(data, "srcset");
```

Comment:
The `<source>` tag supports:

- [`src` for `<audio>` or `<video>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/source#src)
- [`srcset` for `<picture>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/source#srcset)

Ideally we should support both, but I'm not familiar with how these functions work in the context of the parser. It seems that it mutates the `data` as it gets passed around, so I'm not sure how safe it is to e.g. store the original `pos` and call `checkAttribute` for `src` with the original `pos` after checking for `srcset`.

Reply:
Yeah, the amount of random state that this code needs to parse around is very unfortunate; it's gotten worse since the original implementation because markdown is *special*... I *think* that should be fine here; but I really need to revisit this at some point.

Unfortunately, doing that isn't good enough. `srcset` isn't just one `src`, it's a potentially comma separated list with additional metadata ([spec](https://html.spec.whatwg.org/multipage/images.html#srcset-attributes))